action.yml

```diff
@@ -40,6 +40,10 @@ inputs:
     description: 'Fixed hostname to use.'
     required: false
     default: ''
+  statedir:
+    description: 'Optional state directory to use (if unset, memory state is used)'
+    required: false
+    default: ''
```

Comment:
maybe set the default `mem:` value here, so you don't need the if/else below?

Reply:
`--state=mem:` is a different flag to `--statedir=` so it still wouldn't work without a check.

Reply:
doh!  I totally overlooked that.  Disregard then 😛

Reply:
oh, I know what I was thinking that... you can just set `--state` to a file and then `--statedir` will be derived from that.  That's what I normally do in testing.  But having `statedir` be the config option here makes more sense.

Reply:
While I haven't tested, I think that these may be different?
<https://tailscale.com/kb/1278/tailscaled#flags-to-tailscaled>

> To control where state (including preferences and keys) is stored, use one of:
> 
>    --statedir= for a directory on disk where to store config, keys, Taildrop files, and other state.
>     --state= to either a /path/to/file, kube:<secret-name> to use Kubernetes secrets, arn:aws:ssm:... to store state in AWS SSM, or mem: to not store state and reigster as an ephemeral node. By default, if not provided, the state is stored in <statedir>/tailscaled.state.

`--state` is specifically for the `tailscaled.state` file, but `--statedir` gives a whole directory for config/keys which is what is needed for Taillnet Lock to work (it needs somewhere to store keys).

Reply:
And I properly read what you mean now with
> and then --statedir will be derived from that. 

This isn't documented but if it works and is supposed to work I'm on board!

Reply:
looks like the KB is missing a little bit of the documentation we have in the CLI...

```
% tailscaled --help

...
  -statedir string
        path to directory for storage of config state, TLS certs, temporary incoming Taildrop files, etc. 
        If empty, it's derived from --state when possible.
```

Reply:
To make use of that in a way that doesn't have a check in code I think you would have to change the option to be a state file rather than a directory.

e.g. if you set `state: /tmp/tailscale/tailscaled.sock` then you can safely set that and have the default as `mem:`

But if you expose the actions option as `statedir: /tmp/tailscale/` with a default of `mem:` then you still need to think "is this set to `mem:`" before you go setting the `--state` file.

Reply:
yeah, I had originally just forgotten there are two different flags.  I think this PR is good to merge as-is.